src/main/java/com/microsoft/sqlserver/jdbc/SQLCollation.java

```diff
@@ -602,7 +600,23 @@ final Encoding checkSupported() throws UnsupportedEncodingException
         return this;
     }
 
-    final String charsetName() { return charsetName; }
+    final Charset charset() throws SQLServerException
+    {
+        try
+        {
+            checkSupported();
+            if (charset == null)
+            {
+                charset = Charset.forName(charsetName);
```

Comment:
@marschall thank you for your feedback.
I totally agree to what you said, we definitely need to look up once.

However, what I meant was something like this.:
```
        try
    	{
    		checkSupported();
    	} 
    	catch (UnsupportedEncodingException e) 
              ...
    	return Charset.forName(charsetName);
```
the look up is already called within `checkSupported` (by `Charset.isSupported()`)
I am not sure why we need the member variable, or why we need to check if the variable is null.